src/main/java/org/sonar/plugins/stash/issue/MarkdownPrinter.java

```diff
@@ -40,40 +42,45 @@ public static String printIssueListBySeverityMarkdown(SonarQubeIssuesReport repo
     StringBuilder sb = new StringBuilder();
 
     Map<String, SonarQubeIssue> rules = report.getUniqueRulesBySeverity(severity);
-    for (SonarQubeIssue issue : rules.values()) {
-      sb.append("| ").append(MarkdownPrinter.printIssueMarkdown(issue, sonarQubeURL)).append(" |").append(NEW_LINE);
+    for (String rule : rules.keySet()) {
+      SonarQubeIssue issue = rules.get(rule);
+      sb.append("| ").append(issue.printIssueMarkdown(sonarQubeURL)).append(" |").append(NEW_LINE);
     }
 
     return sb.toString();
   }
-
-  /**
-   * Get issue report.
-   */
-  public static String printReportMarkdown(SonarQubeIssuesReport report, String sonarQubeURL, int issueThreshold) {
+  
+  public static String printReportMarkdown(String stashProject, String stashRepo, String pullRequestId,

```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Method has 8 parameters, which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00107)